pkg/releaseutil/kind_sorter.go

```diff
@@ -67,47 +62,34 @@ var InstallOrder KindSortOrder = []string{
 	"APIService",
 }
 
+// addListKind adds the List equivalent for each resource kind in the list.
+func addListKind(resources KindSortOrder) KindSortOrder {
```

Comment:
Is it guaranteed that all Kubernetes tyes have a corresponding "List" type? Offhand, I would guess it is. But there may be exceptions?

And does it actually matter if we generate an "invalid" *List type here?

Reply:
Different versions of Kubernetes support different versions of the K8S resources. Older versions may not necessarily support list types for all of them.
However, this doesn't matter. Whether a list type is used or not is up to the author of a Helm template. 

All that is done in this part of the code is to make sure that templates are ordered correctly. To make sure that templates / resources that have dependencies are applied in the correct order for Kubernetes to resolve those dependencies.
Without this part Kubernetes may fail to apply the templates.

The core part of this code has been around since the beginning of Helm. Every time someone runs into an ordering problem this has been updated. Over time some list types were added as needed. Since I needed a StatefulsetList, I decided to just automatically add list types to all registered types, so we don't have to do this one by one whenever someone runs into a problem.